### ISSUE TITLE
IPv6 support

### ISSUE THREAD

**Issue by asafyish** [nickdesaulniers/node-nanomsg#63]
Hi,

I am trying to connect to "tcp://[fe80::3e15:c2ff:feb9:67c2]:7777" but getting an error message.
From what I googled, it seems that nanomsg is coming with IPv6 disabled...
Because this project is using a fork of nanomsg, is it possible to enable it ?

**Comment by reqshark** [comment]
you should be able to enable it with a call to `setsockopt()` 

the level would be `NN_SOL_SOCKET`

hmm. you know we ought to make a simple function on the namespace like the rest of the sockopt api

**Comment by nickdesaulniers** [comment]
from the [man pages](http://nanomsg.org/v0.2/nn_setsockopt.3.html):
```
 NN_IPV4ONLY

    If set to 1, only IPv4 addresses are used. If set to 0, both IPv4 and IPv6 addresses are used. The type of the option is int. Default value is 1.
```
It looks like [we don't expose that socket option](https://github.com/nickdesaulniers/node-nanomsg#nanosockettype-options)?  We should also update [the docs](https://github.com/nickdesaulniers/node-nanomsg/blob/master/docs/address_strings.markdown) on tcp address strings.  It might be preferable to always turn this option on, or try and parse the address string and enable this automatically.

**Comment by nickdesaulniers** [comment]
published in 0.4.2, please see https://github.com/nickdesaulniers/node-nanomsg/blob/6df49d2d8f28715bf8877c894a470557ea80a067/test/sockoptapi.js#L73-L76 for usage.  @reqshark , we need to update the readme for this option.
